budgeteer-web-interface/src/main/java/org/wickedsource/budgeteer/web/pages/budgets/edit/form/EditBudgetForm.java

```diff
@@ -72,7 +73,11 @@ private void addComponents() {
             @Override
             protected void onEvent(AjaxRequestTarget target) {
                 if (getModelObject().getTags().size() > 0) {
-                    BudgeteerSession.get().getBudgetFilter().getSelectedTags().remove(getModelObject().getTags().remove(0));
+                    String tag = getModelObject().getTags().remove(0);
+                    BudgetTagFilter filter = BudgeteerSession.get().getBudgetFilter();
+                    if (filter != null) {
```

Comment:
Where does this filter get initialized? The NPE can only occur if the object is created lazy. Did you check other usages of `BudgeteerSession.get().getBudgetFilter()` for potential NPEs?